tv_apps/dashboard/js/digital_clock.js

```diff
@@ -0,0 +1,142 @@
+        window.clearInterval(this.intervalId);
+        this.intervalId = null;
+      }
+    },
+
+    /**
+     * Start ticking the clock.
+     * @param  {Function} refresh A callback to run on every tick.
+     */
+    _start: function (refresh) {
+      var now = new Date();
+
+      this.timeoutId = setTimeout(function () {
+        refresh(new Date());
+
+        this.intervalId = setInterval(function () {
+          refresh(new Date());
+        }, 1000);
+      }.bind(this), 1000 - now.getMilliseconds());
+    },
+
+    /**
+     * Init the time of the clock.
+     */
+    _initTime: function (date) {
+      var hours = date.getHours();
+      var minutes = date.getMinutes();
+      var seconds = date.getSeconds();
+
+      this.elements['hours-tens-digit'].textContent = Math.floor(hours / 10);
+      this.elements['hours-units-digit'].textContent = hours % 10;
+      this.elements['minutes-tens-digit'].textContent =
+        Math.floor(minutes / 10);
+      this.elements['minutes-units-digit'].textContent = minutes % 10;
+      this.elements['seconds-tens-digit'].textContent =
+        Math.floor(seconds / 10);
+      this.elements['seconds-units-digit'].textContent = seconds % 10;
+
+      this._changeDate(date);
+
+      this.date = date;
+    },
+
+    /**
+     * Update the display date and time.
+     * @param  {Date} date The Date to be displayed.
+     */
+    _update: function (date) {
+      var hours = date.getHours();
+      var minutes = date.getMinutes();
+      var seconds = date.getSeconds();
+
+      if (this.date.getHours() !== hours) {
+        this._changeTime('hours', this.date.getHours(), hours);
+      }
+
+      if (this.date.getMinutes() !== minutes) {
+        this._changeTime('minutes', this.date.getMinutes(), minutes);
+      }
+
+      this._changeTime('seconds', this.date.getSeconds(), seconds);
+
+      if (this.date.getDate() !== date.getDate()) {
+        this._changeDate(date);
+      }
+      this.date = date;
+    },
+
+    _changeTime: function(unit, oldValue, newValue) {
+      if (Math.floor(oldValue / 10) !== Math.floor(newValue / 10)) {
+        this._changeDigit(unit + '-tens-digit', Math.floor(newValue / 10));
+      }
+      this._changeDigit(unit + '-units-digit', newValue % 10);
+    },
+
+    /**
+     * Change the specified digit and handle the changing animations.
+     * @param  {String} position The position of the element.
+     * @param  {Number} digit    The number to be changed to.
+     */
+    _changeDigit: function (position, digit) {
+      // TODO: handle animations
+      this.elements[position].textContent = digit;
+    },
+
+    /**
+     * Change the display date using mozL10n.
+     * @param  {[Date} date The date to be displayed.
+     */
+    _changeDate: function (date) {
+      navigator.mozL10n.ready(function() {
+        var dateTimeFormat = new navigator.mozL10n.DateTimeFormat();
+        var format = navigator.mozL10n.get('dateFormat');
```

Comment:
`l10n.get` has been deprecated. We have to use `formatValue` instead
But in face `DateTimeFormat` itself uses `l10n.get` and many of them used in TV, I prefer not to block this here. But I would suggest not to use it.